### PR TITLE
Add delete_range method

### PR DESCRIPTION
**Pull request by mk-fg**
Note that it also changes delete_prefix signature to be consistent with other methods there (prefix -> key_prefix, prev_kv keyword, descriptions in the docstring), and delete_range uses similar signature as well.

### CODE REVIEW

File: tests/integration/test_kv.py [hunk 1]
```diff
@@ -233,6 +233,65 @@ async def test_delete_prefix_keys(etcdctl, etcd):
...
+    assert response.deleted == 2
+
+    kv1, kv2 = response.prev_kvs
+    assert kv1.value == b'value1'
+    assert kv2.value == b'value2'
+
+
+@pytest.mark.asyncio
+async def test_delete_range_keys(etcdctl, etcd):
+    key_range = {'/key\1': 'value0',
```
Comment (martyanov) [inline tests/integration/test_kv.py]: Could you please reformat  the dictionary this way?
```
key_range = {
    '/key\1': 'value0',
    '/key1': 'value1',
}
```

File: tests/integration/test_kv.py [hunk 2]
```diff
@@ -233,6 +233,65 @@ async def test_delete_prefix_keys(etcdctl, etcd):
...
+
+    result = await etcd.get(b'/key3\1')
+    assert result is None
+
+    result = await etcd.get(b'/key4')
+    assert result.value == b'value4'
+
+
+@pytest.mark.asyncio
+async def test_delete_range_keys_prev_kv(etcdctl, etcd):
```
Comment (martyanov) [inline tests/integration/test_kv.py]: `test_delete_range_keys_with_prev_kv` for consistency.

File: tests/integration/test_kv.py (hunk 1 above)
Comment (martyanov) [reply tests/integration/test_kv.py]: Also, `key_range` isn't used anywhere else, so may be avoid the variable declaration.

```python
for key, value in {
    'key': 'value',
}:
    etcdctl('put', key, value)
```

**Review by martyanov — CHANGES_REQUESTED** [review]

**Comment by mk-fg** [comment]
Should be all fixed up.

**Comment by martyanov** [comment]
@mk-fg  could you please rebase the branch over upstream master?

**Review by martyanov — APPROVED** [review]

**Comment by mk-fg** [comment]
Done.

**Comment by martyanov** [comment]
Cherry-picked in https://github.com/martyanov/aetcd/commit/d7de6049bf5b7dc73e71cb988de0d180a93348f0.

**Pull request closed without merging**